### PR TITLE
Connect Migration App to Import / Export

### PR DESCRIPTION
**Pull request by cgardens**
## What
* Adjust export of db to serialize json fields as json instead of a string.
* Handle differences in naming and file structure
* Fix issue when accessing same resources multiple times (`MoreResources`)
* Handle extracting and writing tarballs. (includes separating arg parsing and archiving into a runner class)

### CODE REVIEW

File: airbyte-server/src/main/java/io/airbyte/server/handlers/ArchiveHandler.java [hunk 1]
```diff
@@ -72,7 +72,7 @@ public ArchiveHandler(final String version, final ConfigRepository configReposit
 
   public File exportData() {
     try {
-      final Path tempFolder = Files.createTempDirectory(ARCHIVE_FILE_NAME);
+      final Path tempFolder = Files.createTempDirectory(Path.of("/tmp"), ARCHIVE_FILE_NAME);
```
Comment (cgardens) [inline airbyte-server/src/main/java/io/airbyte/server/handlers/ArchiveHandler.java:75]: did this mostly so i could explore these files easily.

File: airbyte-migration/src/main/java/io/airbyte/migrate/migrations/MigrationV0_11_0.java [hunk 2]
```diff
@@ -46,7 +46,7 @@
 
   @Override
   public String getVersion() {
-    return "v0.11.0-alpha";
+    return "0.10.0-alpha";
```
Comment (ChristopheDuong) [inline airbyte-migration/src/main/java/io/airbyte/migrate/migrations/MigrationV0_11_0.java:49]: Why is this down to 0.10.0 when the class is V0_11_0?

File: airbyte-server/src/main/java/io/airbyte/server/converters/DatabaseArchiver.java [hunk 3]
```diff
@@ -96,7 +100,14 @@ private void writeTableToArchive(Table<?> table) throws Exception {
         .from(table.getName())
         .fetchStream());
     records.forEach(r -> Exceptions.toRuntime(() -> {
+      final Set<String> jsonFieldNames = Arrays.stream(r.fields())
+          .filter(f -> f.getDataType().getTypeName().equals("jsonb"))
+          .map(Field::getName)
+          .collect(Collectors.toSet());
       final JsonNode row = Jsons.deserialize(r.formatJSON(DB_JSON_FORMAT));
+      // for json fields, deserialize them so they are treated as objects instead of strings. this is to
+      // get around that formatJson doesn't handle deserializing them for us.
```
Comment (ChristopheDuong) [inline airbyte-server/src/main/java/io/airbyte/server/converters/DatabaseArchiver.java:109]: Cool!

File: airbyte-server/src/main/java/io/airbyte/server/handlers/ArchiveHandler.java (hunk 1 above)
Comment (ChristopheDuong) [reply airbyte-server/src/main/java/io/airbyte/server/handlers/ArchiveHandler.java:75]: Are we reverting these back once we are done?

Comment (cgardens) [reply airbyte-server/src/main/java/io/airbyte/server/handlers/ArchiveHandler.java:75]: i wasn't planning to. it's not bad practice to explicitly put it in `/tmp`

File: airbyte-migration/src/main/java/io/airbyte/migrate/migrations/MigrationV0_11_0.java (hunk 2 above)
Comment (cgardens) [reply airbyte-migration/src/main/java/io/airbyte/migrate/migrations/MigrationV0_11_0.java:49]: getVersion and the class name should match. that's a mistake. as for the version itself, that's going to to be fluid until we "officially" release the migration stuff.

**Review by ChristopheDuong — APPROVED** [review]

**Pull request merged**